nexus_constructor/main_window.py

```diff
@@ -156,8 +156,11 @@ def save_to_filewriter_json(self):
         filename = file_dialog(True, "Save Filewriter JSON File", JSON_FILE_TYPES)
 
         if filename:
+            if not (len(filename) > 5 and filename[-5:].lower() == ".json"):
```

Comment:
```suggestion
            if not filename.endswith(".json"):
```

Reply:
```suggestion
            if not filename.lower().endswith(".json"):
```

Reply:
To be on the safe side. The string is immutable anyway so filename should not be impacted.